protocol/x/vault/keeper/orders_test.go

```diff
@@ -287,33 +293,115 @@ func TestRefreshVaultClobOrders(t *testing.T) {
 				)
 				return genesis
 			}).Build()
-			ctx := tApp.InitChain().WithIsCheckTx(false)
-
-			// Check that there's no stateful orders yet.
-			allStatefulOrders := tApp.App.ClobKeeper.GetAllStatefulOrders(ctx)
-			require.Len(t, allStatefulOrders, 0)
+			ctx := tApp.InitChain()
 
-			// Refresh vault orders.
-			err := tApp.App.VaultKeeper.RefreshVaultClobOrders(ctx, tc.vaultId)
-			allStatefulOrders = tApp.App.ClobKeeper.GetAllStatefulOrders(ctx)
 			if tc.expectedErr != nil {
-				// Check that the error is as expected.
-				require.ErrorContains(t, err, tc.expectedErr.Error())
-				// Check that there's no stateful orders.
-				require.Len(t, allStatefulOrders, 0)
+				// Verify that no order is placed and chain doesn't halt.
+				require.Empty(t, tApp.App.ClobKeeper.GetAllStatefulOrders(ctx))
+				tApp.AdvanceToBlock(uint32(tApp.GetBlockHeight())+1, testapp.AdvanceToBlockOptions{})
 				return
-			} else {
-				// Check that there's no error.
-				require.NoError(t, err)
-				// Check that the number of orders is as expected.
-				params := tApp.App.VaultKeeper.GetParams(ctx)
-				require.Len(t, allStatefulOrders, int(params.Layers*2))
-				// Check that the orders are as expected.
-				expectedOrders, err := tApp.App.VaultKeeper.GetVaultClobOrders(ctx, tc.vaultId)
-				require.NoError(t, err)
-				for i := uint32(0); i < params.Layers*2; i++ {
-					require.Equal(t, *expectedOrders[i], allStatefulOrders[i])
-				}
+			}
+
+			// Vault should place its initial orders.
```

Comment:
nit: Add a check after each block that `mostRecentClientIds` of the vault is the same as all the stateful orders.